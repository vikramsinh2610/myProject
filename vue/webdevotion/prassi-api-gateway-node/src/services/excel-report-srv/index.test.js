const xlsx = require('xlsx');
const { excelReport } = require('.');

const data = {
  headers: [
    { field: 'name', position: 1, translation: 'Nome' },
    { field: 'city', position: 3, translation: 'Город' },
    { field: 'age', position: 2, translation: 'Età' }
  ],
  data: [
    { name: 'Luca', age: '22', city: 'Milano' },
    { name: 'Joe', city: 'New York', age: '23' },
    { age: '23', city: 'New York' },
  ]
};

describe('Excel report service', () => {
  test('should create buffer properly',
    () => {
      const buffer = excelReport(data);
      const parsedBuffer = xlsx.read(buffer);
      return expect(parsedBuffer.Sheets).toMatchSnapshot();
    });
});
