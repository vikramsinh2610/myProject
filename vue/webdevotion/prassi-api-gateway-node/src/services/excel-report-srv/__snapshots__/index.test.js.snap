// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Excel report service should create buffer properly 1`] = `
Object {
  "Report": Object {
    "!ref": "A1:C4",
    "A1": Object {
      "h": "Nome",
      "t": "s",
      "v": "Nome",
      "w": "Nome",
    },
    "A2": Object {
      "h": "Luca",
      "t": "s",
      "v": "Luca",
      "w": "Luca",
    },
    "A3": Object {
      "h": "Joe",
      "t": "s",
      "v": "Joe",
      "w": "Joe",
    },
    "B1": Object {
      "h": "Età",
      "t": "s",
      "v": "Età",
      "w": "Età",
    },
    "B2": Object {
      "h": "22",
      "t": "s",
      "v": "22",
      "w": "22",
    },
    "B3": Object {
      "h": "23",
      "t": "s",
      "v": "23",
      "w": "23",
    },
    "B4": Object {
      "h": "23",
      "t": "s",
      "v": "23",
      "w": "23",
    },
    "C1": Object {
      "h": "Город",
      "t": "s",
      "v": "Город",
      "w": "Город",
    },
    "C2": Object {
      "h": "Milano",
      "t": "s",
      "v": "Milano",
      "w": "Milano",
    },
    "C3": Object {
      "h": "New York",
      "t": "s",
      "v": "New York",
      "w": "New York",
    },
    "C4": Object {
      "h": "New York",
      "t": "s",
      "v": "New York",
      "w": "New York",
    },
  },
}
`;
